add bank/dialog textures

sgfost opened the pull request:
adds sprite overrides for general dialog box background+close button and other bank elements

melkypie commented on src/main/java/melky/resourcepacks/SpriteOverride.java:
```diff
@@ -160,6 +165,9 @@
 	SCROLLBAR_THUMB_MIDDLE_DARK(SpriteID.SCROLLBAR_THUMB_MIDDLE_DARK, Folder.SCROLLBAR),
 
+    WINDOW_CLOSE_BUTTON(SpriteID.WINDOW_CLOSE_BUTTON, Folder.OTHER),
+	WINDOW_CLOSE_BUTTON_HOVERED(SpriteID.WINDOW_CLOSE_BUTTON_HOVERED, Folder.OTHER),
```
These both should be moved between `COMPASS` and `WINDOW_CLOSE_BUTTON_RED_X`. Also the `WINDOW_CLOSE_BUTTON`, needs to use a single tab for alignment rather than spaces

melkypie commented:
Also where in game is the `DIALOG_BACKGROUND` located?

melkypie commented on src/main/java/melky/resourcepacks/SpriteOverride.java:
```diff
@@ -136,6 +136,11 @@
+	DIALOG_BACKGROUND(SpriteID.DIALOG_BACKGROUND, Folder.OTHER),
+	BANK_TAB(SpriteID.BANK_TAB, Folder.OTHER),
+	BANK_TAB_HOVERED(SpriteID.BANK_TAB_HOVERED, Folder.OTHER),
+	BANK_TAB_SELECTED(SpriteID.BANK_TAB_SELECTED, Folder.OTHER),
```
I'd rather have these in a new BANK folder, in there is more bank stuff to add and not to spam the other folder

sgfost commented:
Ok I fixed everything, and for completeness added sprites for borders of dialog boxes. 

`DIALOG_BACKGROUND` is the background for most dialog boxes in the game i.e. the bank, deposit boxes, raids sign up, etc. 
`IRON_RIVETS_*` and `BOTTOM_LINE_MODE_*` are the borders/accents for dialog boxes.

examples of these if you are wondering:
[iron rivets borders/dialog box bg](https://i.imgur.com/sp6M7IO.png)
[bottom line mode borders/dialog box bg](https://i.imgur.com/OrkuWb8.png)

Added folders `bank` and `dialog_boxes`to keep organized

melkypie commented on src/main/java/melky/resourcepacks/SpriteOverride.java:
```diff
@@ -136,6 +140,31 @@
+	IRON_RIVETS_CORNER_BOTTOM_LEFT(SpriteID.IRON_RIVETS_CORNER_BOTTOM_LEFT, Folder.DIALOG_BOXES),
+	IRON_RIVETS_CORNER_BOTTOM_RIGHT(SpriteID.IRON_RIVETS_CORNER_BOTTOM_RIGHT, Folder.DIALOG_BOXES),
+	IRON_RIVETS_EDGE_TOP(SpriteID.IRON_RIVETS_EDGE_TOP, Folder.DIALOG_BOXES),
+	IRON_RIVETS_EDGE_RIGHT(SpriteID.IRON_RIVETS_EDGE_RIGHT, Folder.DIALOG_BOXES),
```
All of these should include the folder name at the start of them. This is done like so because there can't be duplicate enum entries. So if fixed_mode folder has edge_left and resizeable_mode has the same edge_left they wouldn't get marked as duplicates in the enum

sgfost commented:
Fixed the naming issue (changed `DIALOG_BOXES` to `DIALOG` for brevity).

Also fixed a bug in the new config where you couldn't use non-empty paths, and a small typo while I was at it.

melkypie commented:
added in 1c4d0a7535d55d1f02e2652462843e3f58d0c36f and 67a76d03f950ade63fecebd46431f240b01e5103
